test/modules/TestResultCoordinator/PartitionReceiverHandler.cs

```diff
@@ -0,0 +1,105 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace TestResultCoordinator
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Threading.Tasks;
+    using Microsoft.Azure.Devices.Edge.ModuleUtil;
+    using Microsoft.Azure.Devices.Edge.Util;
+    using Microsoft.Azure.EventHubs;
+    using Microsoft.Extensions.Logging;
+
+    class PartitionReceiveHandler : IPartitionReceiveHandler
+    {
+        const string DeviceIdPropertyName = "iothub-connection-device-id";
+        const string ModuleIdPropertyName = "iothub-connection-module-id";
+        const string SequenceNumberPropertyName = "sequenceNumber";
+        const string EnqueuedTimePropertyName = "iothub-enqueuedtime";
+        const string BatchIdPropertyName = "batchId";
+        const string TrackingIdPropertyName = "trackingId";
+
+        static readonly ILogger Logger = ModuleUtil.CreateLogger(nameof(PartitionReceiveHandler));
+
+        readonly string deviceId;
+        readonly string trackingId;
+        
+        public PartitionReceiveHandler(string trackingId, string deviceId)
+        {
+            this.trackingId = Preconditions.CheckNonWhiteSpace(trackingId, nameof(trackingId));
+            this.deviceId = Preconditions.CheckNonWhiteSpace(deviceId, nameof(deviceId));
+        }
+
+        public int MaxBatchSize { get; set; }
+
+        public async Task ProcessEventsAsync(IEnumerable<EventData> events)
+        {
+            if (events != null)
+            {
+                foreach (EventData eventData in events)
+                {
+                    eventData.SystemProperties.TryGetValue(TrackingIdPropertyName, out object trackingIdFromEvent);
```

Comment:
this cannot be a string?

Reply:
it doesn't do type conversion; need to get object first and cast to string.